src/arrayMethodJoin.js

```diff
@@ -4,8 +4,32 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joinedArr = [];
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      if (i === this.length - 1) {
+        joinedArr[i] = this[i];
+      } else {
+        if (separator === null) {
+          joinedArr[i] = this[i] + 'null';
+        } else {
+          joinedArr[i] = this[i] + separator;
+        }
+      }
+    }
+
+    let str = '';
+
+    for (const e of joinedArr) {
+      str += e;
+    }
```

Comment:
instead of using loop to join array and convert it to string we can use `join` method